test/models/root_repository_test.rb

```diff
@@ -281,6 +281,18 @@ def test_create_with_invalid_checksum_type
       assert_match 'is not included in the list', @root.errors[:checksum_type][0]
     end
 
+    def test_auto_enabled
+      assert @rhel6_root.redhat?
+      @rhel6_root.auto_enabled = false
+      refute_valid @rhel6_root
+      assert @rhel6_root.errors.key?(:auto_enabled)
+      assert_match 'Auto Enablement may only be set on custom repositories.', @rhel6_root.errors[:auto_enabled].first
+
+      refute @fedora_root.redhat?
+      @fedora_root.auto_enabled = false
+      assert_valid @fedora_root
```

Comment:
Should extract this into a second test.  And the test names should tell what is being tested: `test_auto_enabled_invalid_on_redhat_repo` and `test_auto_enabled_valid_on_custom_repo`

Reply:
updated